backend/internal/user/service.go

```diff
@@ -391,6 +393,138 @@ func (us *userService) UpdateUser(userID string, user *User) (*User, *serviceerr
 	return user, nil
 }
 
+// UpdateUserAttributes updates only the attributes of a user while preserving immutable fields.
+func (us *userService) UpdateUserAttributes(
+	userID string, attributes json.RawMessage,
+) (*User, *serviceerror.ServiceError) {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, loggerComponentName))
+	logger.Debug("Updating user attributes", log.String("id", userID))
+
+	if strings.TrimSpace(userID) == "" {
+		return nil, &ErrorMissingUserID
+	}
+
+	if len(attributes) == 0 {
+		return nil, &ErrorInvalidRequestFormat
+	}
+
+	if hasCredentials, svcErr := us.containsCredentialFields(attributes); svcErr != nil {
+		return nil, svcErr
+	} else if hasCredentials {
+		return nil, &ErrorInvalidRequestFormat
+	}
```

Comment:
Shall we follow this to keep it consistent 
```suggestion
     hasCredentials, svcErr := us.containsCredentialFields(attributes);
	if svcErr != nil {
		return nil, svcErr
	}
	if hasCredentials {
		return nil, &ErrorInvalidRequestFormat
	}
```

Reply:
Fixed with https://github.com/asgardeo/thunder/compare/c92ad5aad06073fadc8cf1d49a085fdfa1fc0a09..0e2db2d5848c9b7c2da3ec95ea8604432e701ae3